assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,338 @@
+    except Exception as e:
+        raise AzureMLException._with_error(
+            AzureMLError.create(DeploymentCreationError, exception=e)
+        )
+
+    logger.info("Deployment successful.")
+
+    # set the deployment as default
+    try:
+        logger.info(f"Updating endpoint to make {deployment_name} as default deployment")
+        endpoint = ml_client.batch_endpoints.get(endpoint_name)
+        endpoint.defaults.deployment_name = deployment_name
+        ml_client.begin_create_or_update(endpoint).wait()
+
+        endpoint = ml_client.batch_endpoints.get(endpoint_name)
+    except Exception as e:
+        error_msg = f"Error occured while updating deployment - {e}"
+        raise Exception(error_msg)
+
+    logger.info(f"The default deployment is {endpoint.defaults.deployment_name}")
+    return endpoint, deployment
+
+
+@swallow_all_exceptions(logger)
+def main():
+    """Run main function."""
+    args = parse_args()
+    ml_client = get_mlclient()
+    # get registered model id
+    if args.registration_details:
+        model_info = {}
+        with open(args.registration_details) as f:
+            model_info = json.load(f)
+        model_id = model_info["id"]
+        model_name = model_info["name"]
+    elif args.model_id:
+        model_id = str(args.model_id)
+        model_name = model_id.split("/")[-3]
+    else:
+        raise Exception("Arguments model_id and registration_details both are missing.")
+
+    # Endpoint has following restrictions:
+    # 1. Name must begin with lowercase letter
+    # 2. Followed by lowercase letters, hyphen or numbers
+    # 3. End with a lowercase letter or number
+
+    # 1. Replace underscores and slashes by hyphens and convert them to lower case.
+    # 2. Take 21 chars from model name and append '-' & timstamp(10chars) to it
+
+    endpoint_name = re.sub("[^A-Za-z0-9]", "-", model_name).lower()[:21]
+    endpoint_name = f"{endpoint_name}-{int(time.time())}"
+    endpoint_name = endpoint_name
+
+    endpoint_name = args.endpoint_name if args.endpoint_name else endpoint_name
+    deployment_name = args.deployment_name if args.deployment_name else "default"
+
+    endpoint, deployment = create_endpoint_and_deployment(
+        ml_client=ml_client,
+        endpoint_name=endpoint_name,
+        deployment_name=deployment_name,
+        model_id=model_id,
+        args=args
+    )
+
+    if args.inference_payload:
+        print("Invoking inference with test payload ...")
+        try:
+            input = Input(type="uri_folder", path=rf"{args.inference_payload}")
+
+            job = ml_client.batch_endpoints.invoke(
+                endpoint_name=endpoint.name, input=input
+            )
+
+            ml_client.jobs.stream(job.name)
+
+            scoring_job = list(ml_client.jobs.list(parent_job_name=job.name))[0]
+
+            ml_client.jobs.download(
+                name=scoring_job.name, download_path=".", output_name="score"
```

Comment:
should we also have prediction.csv as output to component post download?

Reply:
As discussed will add in future versions